vertx-sql-client/src/main/java/io/vertx/sqlclient/data/NullValue.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.sqlclient.data;
+
+import io.vertx.codegen.annotations.GenIgnore;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.sqlclient.impl.data.NullValueImpl;
+
+import java.math.BigDecimal;
+import java.time.*;
+import java.time.temporal.Temporal;
+import java.util.UUID;
+
+/**
+ * Instances of this class indicate the type associated with a {@code NULL} column value.
+ * This is useful when infering database types from {@link io.vertx.sqlclient.Tuple} items.
+ */
+@VertxGen
+public interface NullValue {
```

Comment:
Why?

Reply:
>I think we actually must, otherwise it will not work for Vert.x RX (and
similar approaches).

>With Vert.x RX (or any other similar wrapper approach), then NullValue will
not be unwrapped by the internal class and it will use the RX version of
NullValue instead of the original value.

Right. Let me try this approach

Reply:
See new commit